include/na/zoned/layout_synthesizer/router/IndependentSetRouter.hpp

```diff
@@ -96,15 +238,99 @@ class IndependentSetRouter : public RouterBase {
+                       const std::tuple<size_t, size_t, size_t, size_t>& w)
+      -> bool;
+  /**
+   * @brief This struct indicates whether two movements are strictly compatible,
+   * relaxed compatible together with the corresponding merging cost, or
+   * (completely) incompatible.
+   */
+  struct MovementCompatibility {
+    enum class Status : uint8_t {
+      /**
+       * @brief The movements are strictly compatible.
+       * @details The atoms remain on the same row (column) and maintain their
+       * relative/topological order during the movement.
+       */
+      StrictlyCompatible,
+      /**
+       * @brief The movements are compatible with respect to the relaxed routing
+       * constraints.
+       * @details The moved atoms must still remain on the same row (column) but
+       * may change their relative/topological order during the movement
+       * achieved by offsets during pick-up and drop-off.
+       */
+      RelaxedCompatible,
+      /**
+       * @brief The movements are incompatible.
+       * @details The atoms starting in one row (column) do not end up in the
+       * same row (column).
+       */
+      Incompatible
+    };
+
+    /// Indicates the type of compatibility
+    Status status = Status::Incompatible;
+    /**
+     * @brief In the case of `RelaxedCompatible`, the cost to merge the two
+     * movements.
+     * @details The cost is calculated such that it represents the extra time
+     * the offset takes to shift the loaded atoms to deal with the relaxed
+     * routing constraints. More precisely, the cost is proportional to the
+     * cubed time for the sake of easier computation because then it is just
+     * proportional to the distance of the offset. If only one offset is
+     * required, i.e., either horizontal or vertical, the cost is the raw
+     * distance of the offset. If both offsets are required, then the cost is
+     * calculated as the sum of the third roots of the individual distances and
+     * then cubed again.
+     * Hence, the cost must always be a non-negative number.
+     */
+    std::optional<double> mergeCost = std::nullopt;
+
+    /// Factory methods for strict compatibility
+    [[nodiscard]] static auto strictlyCompatible() -> MovementCompatibility {
+      return {.status = Status::StrictlyCompatible, .mergeCost = std::nullopt};
+    }
+
+    /// Factory method for incompatibility
+    [[nodiscard]] static auto incompatible() -> MovementCompatibility {
+      return {.status = Status::Incompatible, .mergeCost = std::nullopt};
+    }
+
+    [[nodiscard]] static auto relaxedCompatible(const double cost)
+        -> MovementCompatibility {
+      return {.status = Status::RelaxedCompatible, .mergeCost = cost};
+    }
+  };
+  /**
+   * Check whether two movements are incompatible with respect to the relaxed
+   * routing constraints, i.e., moved atoms remain not on the same row (column).
```

Comment:
```suggestion
   * Check whether two movements are compatible with respect to the relaxed
   * routing constraints, i.e., moved atoms remain on the same row (column).
```
To better align with the function name